File: app/src/main/java/com/dynamo/spacex/viewmodels/LaunchDetailsViewModel.java
package com.dynamo.spacex.viewmodels;

import androidx.databinding.ObservableField;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.dynamo.spacex.shared.LaunchDetailsUseCase;
import com.dynamo.spacex.shared.TransientDataProvider;

import javax.inject.Inject;

public class LaunchDetailsViewModel extends ViewModel {

    public ObservableField<String> launchDetailsTitle = new ObservableField<>();
    public ObservableField<String> rocketName = new ObservableField<>();
    public ObservableField<String> launchDetailsText = new ObservableField<>();
    public MutableLiveData<String> youtubeVideoId = new MutableLiveData<>();

    private final TransientDataProvider transientDataProvider;

    @Inject
    public LaunchDetailsViewModel(TransientDataProvider transientDataProvider) {
        this.transientDataProvider = transientDataProvider;
        getValuesFromUseCase();
    }

    private void getValuesFromUseCase() {
        LaunchDetailsUseCase useCase = transientDataProvider.getUseCase(LaunchDetailsUseCase.class);
        launchDetailsTitle.set(useCase.getTitle());
        rocketName.set(useCase.getRocketName());
        launchDetailsText.set(useCase.getLaunchDetailsText());
        youtubeVideoId.setValue(useCase.getYoutubeVideoId());
    }
}

File: app/src/main/java/com/dynamo/spacex/viewmodels/LaunchDetailsViewModel.kt
package com.dynamo.spacex.viewmodels

import androidx.databinding.ObservableField
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.dynamo.spacex.shared.LaunchDetailsUseCase
import com.dynamo.spacex.shared.TransientDataProvider

class LaunchDetailsViewModel (
        private val transientDataProvider: TransientDataProvider
): ViewModel() {
    val launchDetailsTitle: ObservableField<String> = ObservableField()
    val rocketName: ObservableField<String> = ObservableField()
    val launchDetailsText: ObservableField<String> = ObservableField()
    val youtubeVideoId = MutableLiveData<String>()

    fun getValuesFromUseCase() {
        if (transientDataProvider.containsUseCase(LaunchDetailsUseCase::class.java)) {
            val useCase = transientDataProvider.getUseCase(LaunchDetailsUseCase::class.java) as LaunchDetailsUseCase
            launchDetailsTitle.set(useCase.title)
            rocketName.set(useCase.rocketName)
            launchDetailsText.set(useCase.launchDetailsText)
            youtubeVideoId.postValue(useCase.youtubeVideoId)
        }
    }
}